eng/pipelines/templates/steps/archetype-sdk-docs.yml

```diff
@@ -0,0 +1,133 @@
+steps:
+  - pwsh: |
+      Invoke-WebRequest -MaximumRetryCount 10 -Uri "https://github.com/mono/api-doc-tools/releases/download/mdoc-5.7.4.9/mdoc-5.7.4.9.zip" `
+      -OutFile "mdoc.zip" | Wait-Process; Expand-Archive -Path "mdoc.zip" -DestinationPath "./mdoc/"
+    workingDirectory: $(Build.BinariesDirectory)
+    displayName: Download and Extract mdoc Zip
+  - pwsh: |
+      Invoke-WebRequest -MaximumRetryCount 10 -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
+      -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "./docfx/"
+    workingDirectory: $(Build.BinariesDirectory)
+    displayName: Download and Extract DocFX
+  - pwsh: |
+      dotnet restore "$(Build.SourcesDirectory)/eng/docgeneration/assets/docgen.csproj" /p:BuildBinariesDirectory=$(Build.BinariesDirectory)
+    displayName: Restore Nuget Packages Needed
+  - task: DotNetCoreInstaller@2
+    displayName: "Use .NET Core sdk $(DotNetCoreSDKVersion)"
+    inputs:
+      version: $(DotNetCoreSDKVersion)
+  - ${{ if ne(variables['Build.DefinitionName'], 'net - client - ci')}}:
+    - ${{ each artifact in parameters.Artifacts }}:
+      - pwsh: |
+          echo "##vso[task.setvariable variable=packageLocation]${{parameters.ServiceDirectory}}/${{artifact.name}}"
+          echo "##vso[task.setvariable variable=artifactSafeName]${{artifact.safeName}}"
+          if ('${{parameters.ServiceDirectory}}' -eq '*') {
```

Comment:
It would be better to use the same pattern as cogservices here and set a directoryName in the core project and then update this logic to:

```
$packageDir= "${{ artifact.directoryName }}"
if ($packageDir -eq "") { $packageLocation = "${{parameters.ServiceDirectory}}" }

echo "##vso[task.setvariable variable=packageLocation]$packageDir/${{artifact.name}}"
```

Probably do the same thing for mgmt to avoid multiple ways to configure this.

Reply:
I assume this will be part of your clean-up work item?